__app__/crop/ingress_hyper.py

```diff
@@ -0,0 +1,211 @@
+"""
+Python module to import data using the Hyper.ag API
+"""
+
+from collections.abc import Iterable
+import logging
+import time
+from datetime import datetime, timedelta
+import requests
+import pandas as pd
+
+from sqlalchemy import and_
+from sqlalchemy.exc import ProgrammingError
+
+from __app__.crop.db import connect_db, session_open, session_close
+from __app__.crop.structure import (
+    TypeClass,
+    SensorClass,
+    ReadingsAranetTRHClass,
+)
+from __app__.crop.utils import query_result_to_array
+from __app__.crop.constants import CONST_CROP_HYPER_APIKEY, CONST_ARANET_TRH_SENSOR_TYPE
+
+from __app__.crop.ingress import log_upload_event
+from __app__.crop.sensors import get_aranet_trh_sensor_data
+
+CONST_CHECK_URL_PATH = "https://zcf.hyper.ag/api/sites/0/analytics/v3/device_metrics"
```

Comment:
Hi @mhauru! Great progress with the PR.

There's one small thing I wanted to ask you to change (which we incorrectly included in our example):

- `https://zcf.hyper.ag/api/sites/0/...` should be
- `https://zcf.hyper.ag/api/sites/1/...`.

It does not matter right now if we use site 0 or site 1 (both should work), but in the future we might reserve 0 for "offline data".

To elaborate, currently the edge device at the farm pushes all the data directly to the cloud, but we plan on adding the ability to buffer the data offline in case of connectivity issues. In these situations we reserved the 0 for the offline deployment.

Sorry for the confusion! If you use site 1, everything should work just as expected.

Reply:
Thanks Rizo, now fixed.